src/core_modules/capture-core/components/DataEntries/SingleEventRegistrationEntry/DataEntryWrapper/DataEntry/epics/getConvertedNewSingleEvent.js

```diff
@@ -43,6 +43,44 @@ export const getNewEventServerData = (state: ReduxState, formFoundation: RenderF
     };
 };
 
+export const getAddEventEnrollmentServerData = (state: ReduxState,
+    formFoundation: RenderFoundation,
+    formClientValues: Object,
+    mainDataClientValues: Object,
+    completed?: boolean,
+) => {
+    const formServerValues = formFoundation.convertValues(formClientValues, convertToServerValue);
+    const mainDataServerValues: Object = convertMainEventClientToServer(mainDataClientValues);
+    const { teiId, enrollmentId, programId, orgUnitId } = state.router.location.query;
+
+    if (!mainDataServerValues.status) {
+        mainDataServerValues.status = completed ? 'ACTIVE' : 'COMPLETED';
+    }
+    if (mainDataServerValues.status === 'COMPLETED') {
+        mainDataServerValues.completedDate = getFormattedStringFromMomentUsingEuropeanGlyphs(moment());
+    }
+
+    return {
+        events: [
+            {
+                ...mainDataServerValues,
+                program: programId,
```

Comment:
Hey, 
There is a helper named [getServerValuesToSaveFromMainEvent](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/DataEntry/common/convertEventToServerValues.js#L5) inside `DataEntry` component. Do you think it can be reused to map the keys from client to server shape?

Thanks!

Reply:
I don't want to mess up with the main event so I made a separate function